tests/tests_sound_composer/test_sound_composer_source_harmonics_two_parameters.py

```diff
@@ -0,0 +1,672 @@
+    f_source_control.time_freq_support = support
+
+    # Create a second SourceControlTime object.
+    source_control2 = SourceControlTime()
+    source_control2.control = f_source_control
+
+    # Create a SourceHarmonicsTwoParameters object test source file and created source
+    # controls.
+    source_obj = SourceHarmonicsTwoParameters(
+        file=pytest.data_path_sound_composer_harmonics_source_2p_in_container,
+        control_rpm=source_control_rpm,
+        control2=source_control2,
+    )
+
+    source_obj.process(sampling_frequency=44100.0)
+    f_output = source_obj.get_output()
+    assert isinstance(f_output, Field)
+    assert len(f_output.data) / 44100.0 == pytest.approx(3.0)
+
+    # Compute the power spectral density over the output signal.
+    psd = PowerSpectralDensity(
+        input_signal=f_output,
+        fft_size=8192,
+        window_type="HANN",
+        window_length=8192,
+        overlap=0.75,
+    )
+    psd.process()
+    psd_squared, psd_freq = psd.get_PSD_squared_linear_as_nparray()
+    delat_f = psd_freq[1] - psd_freq[0]
+
+    # Check the sound power level in the octave bands centered at 500, 1000 and 2000 Hz.
+    # Due to the non-deterministic nature of the produced signal, tolerance is set to 3 dB.
+    psd_squared_band = psd_squared[
+        (psd_freq >= 500 * 2 ** (-1 / 2)) & (psd_freq < 500 * 2 ** (1 / 2))
+    ]
+    level = 10 * np.log10(psd_squared_band.sum() * delat_f / REF_ACOUSTIC_POWER)
+    assert level == pytest.approx(EXP_LEVEL_OCTAVE_BAND_500, abs=3.0)
+
+    psd_squared_band = psd_squared[
+        (psd_freq >= 1000 * 2 ** (-1 / 2)) & (psd_freq < 1000 * 2 ** (1 / 2))
+    ]
+    level = 10 * np.log10(psd_squared_band.sum() * delat_f / REF_ACOUSTIC_POWER)
+    assert level == pytest.approx(EXP_LEVEL_OCTAVE_BAND_1000, abs=3.0)
+
+    psd_squared_band = psd_squared[
+        (psd_freq >= 2000 * 2 ** (-1 / 2)) & (psd_freq < 2000 * 2 ** (1 / 2))
+    ]
+    level = 10 * np.log10(psd_squared_band.sum() * delat_f / REF_ACOUSTIC_POWER)
+    assert level == pytest.approx(EXP_LEVEL_OCTAVE_BAND_2000, abs=3.0)
+
+    # Test output with another source file with inverted controls.
+    source_obj.load_source_harmonics_two_parameters(
+        pytest.data_path_sound_composer_harmonics_source_2p_inverted_controls_in_container,
+    )
+
+    source_obj.process(sampling_frequency=44100.0)
+    f_output = source_obj.get_output()
+    assert isinstance(f_output, Field)
+    assert len(f_output.data) / 44100.0 == pytest.approx(3.0)
+
+    # Compute the power spectral density over the output signal.
+    psd = PowerSpectralDensity(
+        input_signal=f_output,
+        fft_size=8192,
+        window_type="HANN",
+        window_length=8192,
+        overlap=0.75,
+    )
+    psd.process()
+    psd_squared, psd_freq = psd.get_PSD_squared_linear_as_nparray()
+    delat_f = psd_freq[1] - psd_freq[0]
+
+    # Check the sound power level in the octave bands centered at 500, 1000 and 2000 Hz.
```

Comment:
@ansaminard why do we need a 3dB tolerance for H 2 parameters whereas we only have a 1 dB tolerance for H 1 parameter ? (how did you get the ref values?)

Reply:
Reduced to 1 dB

Reply:
@anshlachamb Also to answer your question, I ran the synthesis a few times, took a rough average of obtained values, and added some room to make sure the pipeline does not fail every now and then